sdk/core/src/request_options/content_length.rs

```diff
@@ -5,7 +5,6 @@ use http::request::Builder;
 pub struct ContentLength(i32);
 
 impl ContentLength {
-    /// Create a new `ContentLength`
```

Comment:
Why remove this (other than it being somewhat obvious)?

Reply:
Yes, just because it's obvious. Open to leaving it in though it provides some value I overlooked.

Reply:
I'd personally like if every public function and type were documented even if obvious.

Reply:
I see the value of consistent public docs and might have gone a bit too far removing this, probably because I just encountered [these `PathRenameMode` docs](https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/create#pathrenamemode) during this PR - I still don't know what this parameter does and just copied what the .Net SDK does ;)

However, currently only 1 of the other 30 types (`MaxItemCount`) has such a comment. Maybe we need an issue to add docs to all public functions (maybe per crate)?